src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -350,31 +352,48 @@ class TezosNodeOperator(
       (fetchCurrentQuorum, fetchCurrentProposal).mapN(CurrentVotes.apply)
     }
 
+  /** Fetches proposals for given blocks */
+  def getProposals(blocks: List[BlockData]): Future[List[(BlockHash, Option[ProtocolId])]] = {
```

Comment:
Just wondering: is there a way that we could collect all voting details together (maybe in a preparatory method) and then re-use them wherever they're needed.
I mention this as I see we seem to be fetching the same data many times per cycle, both here and in `getVotingDetails` and `getVotes`?

Do that corresponds to the same data?

Reply:
`getVotingDetails` and `getVotes` are different now - I've noticed that in the only place that method was used most of the values were discarded

Reply:
Do you think we can optimize nonetheless?

Reply:
optimized!

Reply:
luvely